### PR TITLE
Remove reader/comments component and switch my-sites/posts over to the newer block

### PR DESCRIPTION
**Pull request by bluefuton**
The `reader/comments` component has been superceded by the new comments block, `blocks/comments`.

The Reader itself is already using the new block, but My Sites > Posts still uses the older version.

This PR removes the old component and switches My Sites > Posts to use the new block.

screen shot 2017-01-03 at 16 09 41

### To test

Head to http://calypso.localhost:3000/posts/my and ensure that comment threads appear correctly.

### Todo

- [x] switch `my-sites/posts/post.jsx` over to use new comments block
- [x] tidy up `!important` styles in `blocks/comments/style.scss`
- [x] remove `reader/comments` component completely
- [x] turn off "x comments" header for My Sites
- [x] tweak date spacing

Fixes https://github.com/Automattic/wp-calypso/issues/9705.

### CODE REVIEW

**Comment by samouri** [comment]
jeez louise 🔴 🔴 🔴 🔴 👍

File: client/blocks/comments/style.scss [hunk 1]
```diff
@@ -122,7 +122,7 @@
 	margin: 0;
 
 	&.is-root {
-		margin-top: 0 !important; // the !important can go when the old reader/comments is retired
```
Comment (samouri) [inline client/blocks/comments/style.scss]: nice follow through :)

**Comment by samouri** [comment]
Tested:
- [x] built locally
- [x] all okapis in my streams loaded 👍 

I went to my-sites but wasn't sure exactly what to test there :/

**Comment by mtias** [comment]
Thanks!

**Pull request merged**